### PR TITLE
[Vue] Remove AngularJS uses from tracking.js

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR depends on https://github.com/matomo-org/matomo/pull/19348.**

Changes:
* Remove uses of angular from tracking.js.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: javascripts/tracking.js [hunk 1]
```diff
@@ -130,19 +130,22 @@ piwikUsageTracking.createTrackersIfNeeded = function ()
...
 
-        $rootScope.$on('$locationChangeSuccess', function (event, newUrl, oldUrl) {
+        var watch = window.Vue.watch;
+        var MatomoUrl = window.CoreHome.MatomoUrl;
+        watch(
+          function () { return MatomoUrl.url; },
+          function (newUrl, oldUrl) {
             var urlAnonymizer = createUrlAnonymizer();
 
             if (urlAnonymizer.isPiwik3ReportingUrl() && newUrl !== oldUrl) {
```
Comment (sgiehl) [inline javascripts/tracking.js]: I'm not very deep into that plugin, but can't we maybe remove that part of the code completely? As the version released from `5.x-dev` branch will be only compatible with Matomo 5, there shouldn't be any checks needed anymore for urls that might look like old Piwik 3 urls. Or is that somehow used to track the admin part maybe? 🤔

Comment (diosmosis) [reply javascripts/tracking.js]: @sgiehl looks like it's not used, I couldn't see it ever get triggered. I also couldn't see locationChangeSuccess ever get invoked... seems like broadcast.loadAjaxContent isn't really used anymore. Modified it to use the `piwikPageChange` event, and it started tracking pageviews again. Thanks for catching that.

Comment (sgiehl) [reply javascripts/tracking.js]: As we are trying to get rid of the old name over time and we have the possibility to break bc with Matomo 5, wouldn't it make sense to rename the `piwikPageChange` event to `matomoPageChange` in core?

Comment (diosmosis) [reply javascripts/tracking.js]: That's a bit out of scope for vue changes, but I agree Matomo 5 is a good opportunity to change the name. If it changes in another PR in core, I can change it in any vue related prs that are pending.

Comment (diosmosis) [reply javascripts/tracking.js]: ping @sgiehl ^

Comment (sgiehl) [reply javascripts/tracking.js]: @diosmosis I've created https://github.com/matomo-org/matomo/pull/19710, feel free to review and merge that one.

Comment (diosmosis) [reply javascripts/tracking.js]: @sgiehl updated this PR

**Comment by diosmosis** [comment]
@sgiehl can this be merged?

**Comment by sgiehl** [comment]
Just looking at the code, the changes look good to me. But I have actually never used or worked with the plugin. So it would take me a lot time to get familiar with it in the first place, for being able to prove that it still does the same as before. Unfortunately I don't have the time for doing that at the moment. Not sure if @AltamashShaikh is familiar with the plugin and can take over here.

**Review by AltamashShaikh — APPROVED** [review]
@diosmosis I can see its tracking as expected

**Pull request merged**